### PR TITLE
Added User entity to the project

### PR DESCRIPTION
**Pull request by kaiiiseeel**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/entity/User.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@Getter
+@Setter
+@ToString
+@EqualsAndHashCode
+@Table(name = "users")
+public class User {
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/entity/User.java:24]: Soft delete should be applied to user as well

File: src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+        if (userRepository.findAllByEmail(requestDto.getEmail()).isPresent()) {
+            throw new RegistrationException("The user with this email is already registered");
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java]:
```suggestion
            throw new RegistrationException("The user with this email: EMAIL is already registered");

```

File: src/main/java/mate/academy/bookstore/validation/FieldMatchValidator.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Objects;
+import mate.academy.bookstore.dto.UserRegistrationRequestDto;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch,
+        UserRegistrationRequestDto> {
+    @Override
+    public boolean isValid(UserRegistrationRequestDto requestDto,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        return Objects.equals(requestDto.getPassword(), requestDto.getRepeatedPassword());
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/validation/FieldMatchValidator.java]: This annotation works only with password and repeatedPassword, but it should work with 2 any fields, that is defined in annotation properties

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                  constraints:
+                    nullable: false
+              - column:
+                  name: last_name
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: shipping_address
+                  type: varchar(500)
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml]:
```suggestion
                  type: varchar(500)
                  
```

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Good job! Left minor comments

File: src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+        if (userRepository.findAllByEmail(requestDto.getEmail()).isPresent()) {
```
Comment (karma-o) [inline src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java]: a bit more efficient and simple
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```

File: src/main/java/mate/academy/bookstore/validation/FieldMatchValidator.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        firstField = constraintAnnotation.firstFieldName();
+        secondField = constraintAnnotation.secondFieldName();
+    }
+
+    @Override
+    public boolean isValid(Object value,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        Object firstObject = new BeanWrapperImpl(value).getPropertyValue(firstField);
```
Comment (karma-o) [inline src/main/java/mate/academy/bookstore/validation/FieldMatchValidator.java]: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/BeanWrapperImpl.html
> NOTE: As of Spring 2.5, this is - for almost all purposes - an internal class. It is just public in order to allow for access from other framework packages. For standard application access purposes, use the [PropertyAccessorFactory.forBeanPropertyAccess(java.lang.Object)](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/PropertyAccessorFactory.html#forBeanPropertyAccess(java.lang.Object)) factory method instead.

First time seeing this class, so not sure how it works and what it does, had to google it and found out that it is not recommended. overall we could have used simple reflection by getting the class of `value` var and getting a field by `class.getDeclaredField(field)` to then compare the values of those 2 fields.

File: src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 7]
```diff
@@ -43,6 +43,11 @@ public ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundExcept
         return new ResponseEntity<>(exception.getMessage(), HttpStatus.NOT_FOUND);
     }
 
+    @ExceptionHandler(RegistrationException.class)
+    public ResponseEntity<Object> handleRegistrationException(RegistrationException exception) {
+        return new ResponseEntity<>(exception.getMessage(), HttpStatus.CONFLICT);
```
Comment (karma-o) [inline src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java]: if these return plain text, we should change it to some json with timestamp, status phrase and message fields

**Review by karma-o — APPROVED** [review]
GJ! Approved with several comments for improvement

**Pull request merged**